gapic/src/main/com/google/gapid/views/PipelineView.java

```diff
@@ -184,202 +192,293 @@ protected void onUiThreadError(Loadable.Message error) {
   protected void setPipelines(List<API.Pipeline> pipelines) {
     loading.stopLoading();
     disposeAllChildren(stagesContainer);
-    TabFolder folder = createStandardTabFolder(stagesContainer);
-    createPipelineTabs(folder, pipelines);
+    createPipelineTabs(stagesContainer, pipelines);
     stagesContainer.requestLayout();
   }
 
-  private void createPipelineTabs(TabFolder folder, List<API.Pipeline> pipelines) {
+  private void createPipelineTabs(Composite folder, List<API.Pipeline> pipelines) {
+    HashMap<String, Composite> stageMap = new HashMap<String, Composite>();
+
     if (!pipelines.isEmpty()) {
-      for (API.Pipeline result : pipelines) {
-        List<API.Stage> stages = result.getStagesList();
+      Composite stripComposite = withLayoutData( createComposite(folder, new FillLayout(SWT.HORIZONTAL)),
```

Comment:
The strip looks weird when stretched too much on a big screen. Let's not fill the width. Please replace the line with:
```
      RowLayout stripLayout = new RowLayout();
      stripLayout.fill = true;
      stripLayout.spacing = 0;
      stripLayout.wrap = true;
      Composite stripComposite = withLayoutData( createComposite(folder, stripLayout),
```

Reply:
Little worried about wrap being true. Check with Pau about this. I think he may have wanted it to shrink instead. The doc isn't explicit about this.

By the way, unless I implemented one of your changes wrong, this causes the strip to be left justified instead of centered. Also, the boxes are now different sizes, which the mockup does not imply. In fact, these were the reasons I used Fill Layout in the first place. I wasn't entirely pleased with it either, especially since BLEND never quite looked right.

Should we use a grid layout like we did for the KVP grids?

Reply:
The mockups don't address that the layout can be resized, so it is ok to deviate if we have a reason. Left justified is also intended. I will check everything with Pau. No need for grid layout.